packages/@sanity/form-builder/src/sanity/DefaultAssetSource/Asset.tsx

```diff
@@ -124,32 +140,69 @@ export default class Asset extends React.PureComponent<AssetProps, State> {
   }
 
   render() {
-    const {asset, onClick, onKeyPress, isSelected} = this.props
+    const {asset, assetType = 'image', onClick, onKeyPress, isSelected} = this.props
+    const {originalFilename, _id, mimeType, extension, url} = asset
     const {isDeleting, showUsageDialog, deleteError} = this.state
     const imgH = 200 * Math.max(1, DPI)
+    const isImageType = assetType === 'image'
+    const isImageMimetype = mimeType.includes('image')
```

Comment:
edge case for sure, but this should probably check the first part of the mime type? Might give false positives on things like disk images? You could also consider using the `attr-accept` package that we use elsewhere for checking mime type: `accepts('image/*', mimeType)`

Side note: browsers typically only supports a [subset of all possible image types](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types) too, so might be an idea to a) check up front if it can be displayed, or b) only display types of a known format.

Reply:
How would you approach it? Better to just whitelist the ones we know are safe?

Reply:
Maybe if we verify first that the mime type is `image/*`, then we can try to display it with a fallback in case of an error?

Reply:
Thinking some more about this I'm not really sure we want to display images uploaded as files. Since we can't get a thumbnail version for it (they are served from the file cdn, which doesn't support image url parameters), we'd in many cases end up wasting bandwidth on large image files not intended for being viewed. I think it's fair to assume that if an image is uploaded as a file asset it's not intended to be displayed as a regular image - what do you think?

Reply:
In hindsight, I agree with Bjørge on the last comment here. Not showing any previews for images makes it more obvious that you are using the wrong asset type.